src/validation/statement.rs

```diff
@@ -1324,10 +1324,7 @@ fn validate_type_nature<T: AnnotationMap>(
         {
             // check if type_hint and actual_type is compatible
             // should be handled by assignment validation
-            if !(actual_type.has_nature(*generic_nature, context.index)
-                // INT parameter for REAL is allowed
-                | (type_hint.is_real() & actual_type.is_numerical()))
```

Comment:
I think this would cause errors when doing something like IS_VALID(1) instead of IS_VALID(1.0), I'm not sure I didn't test that yet but I believe this is the reason we had this exception here.

Reply:
> I think this would cause errors when doing something like IS_VALID(1) instead of IS_VALID(1.0), I'm not sure I didn't test that yet but I believe this is the reason we had this exception here.

I've tested it, and yes it would cause an `invalid type nature` error.

The example codes and the codes in https://github.com/PLC-lang/oscat/tree/main  have a lot of implicit type castings bewteen `int` and `any_bit`, so I'm also afraid that this adjustment of compiler behavior would cause a lot of refactors on old codes...